build.gradle.kts

```diff
@@ -17,29 +15,39 @@ repositories {
     mavenCentral()
 }
 
-dependencies {
-    api("org.jetbrains.kotlinx:kotlinx-serialization-json:1.4.1")
-    testImplementation(kotlin("stdlib"))
-    testImplementation(kotlin("test"))
-    testImplementation("org.junit.jupiter:junit-jupiter-api:5.9.0")
-    testRuntimeOnly("org.junit.jupiter:junit-jupiter-engine:5.9.0")
-}
-
-java {
-    sourceCompatibility = JavaVersion.VERSION_1_8
-    targetCompatibility = JavaVersion.VERSION_1_8
-    withJavadocJar()
-    withSourcesJar()
-}
+kotlin {
+    jvm {
+        compilations.all {
+            kotlinOptions {
+                jvmTarget = "1.8"
+            }
+        }
+    }
+    linuxX64()
+    mingwX64()
+    macosX64()
 
-tasks {
-    withType<KotlinCompile>().configureEach {
-        this.kotlinOptions {
-            jvmTarget = "1.8"
-            freeCompilerArgs = listOf("-opt-in=kotlinx.serialization.ExperimentalSerializationApi")
+    sourceSets {
+        all {
+            languageSettings.optIn("kotlinx.serialization.ExperimentalSerializationApi")
+        }
+        val commonMain by getting {
+            dependencies {
+                api("org.jetbrains.kotlinx:kotlinx-serialization-json:1.4.1")
+            }
+        }
+        val jvmTest by getting {
+            dependencies {
+                implementation(kotlin("stdlib"))
+                implementation(kotlin("test"))
+                implementation("org.junit.jupiter:junit-jupiter-api:5.9.0")
+                runtimeOnly("org.junit.jupiter:junit-jupiter-engine:5.9.0")
+            }
         }
     }
+}
 
+tasks {
     withType<Javadoc>().configureEach {
```

Comment:
probably it should be rewritten to dokka

Reply:
We have setup for publishing to Github packages in https://github.com/saveourtool/sarif4k/commits/feature/publishing, but Github doesn't require docs (unlike Maven Central)

Yes, dokka, artifact signing and publications should be set up, probably, similarly to how we are doing it in save-cli. However, folks at Detekt can have different conventions for these things, so probably we can simply switch Javadoc tasks with Dokka and suggest these changes as is

Reply:
added dokka, but I'm getting the following message in gradle:
```
> Task :publishSarif4kPublicationToMavenLocal
Multiple publications with coordinates 'io.github.detekt.sarif4k:sarif4k:0.3.0-SNAPSHOT' are published to repository 'mavenLocal'. The publications 'kotlinMultiplatform' in root project 'sarif4k' and 'sarif4k' in root project 'sarif4k' will overwrite each other!
```

And it produces the following artifacts:
```
$ ls ~/.m2/repository/io/github/detekt/sarif4k/*/0.3.0-SNAPSHOT
/c/Users/a00665046/.m2/repository/io/github/detekt/sarif4k/sarif4k/0.3.0-SNAPSHOT:
maven-metadata-local.xml    sarif4k-0.3.0-SNAPSHOT.module  sarif4k-0.3.0-SNAPSHOT-javadoc.jar                   sarif4k-0.3.0-SNAPSHOT-sources.jar
sarif4k-0.3.0-SNAPSHOT.jar  sarif4k-0.3.0-SNAPSHOT.pom     sarif4k-0.3.0-SNAPSHOT-kotlin-tooling-metadata.json

/c/Users/a00665046/.m2/repository/io/github/detekt/sarif4k/sarif4k-jvm/0.3.0-SNAPSHOT:
maven-metadata-local.xml  sarif4k-jvm-0.3.0-SNAPSHOT.jar  sarif4k-jvm-0.3.0-SNAPSHOT.module  sarif4k-jvm-0.3.0-SNAPSHOT.pom  sarif4k-jvm-0.3.0-SNAPSHOT-sources.jar

/c/Users/a00665046/.m2/repository/io/github/detekt/sarif4k/sarif4k-linuxx64/0.3.0-SNAPSHOT:
maven-metadata-local.xml  sarif4k-linuxx64-0.3.0-SNAPSHOT.klib  sarif4k-linuxx64-0.3.0-SNAPSHOT.module  sarif4k-linuxx64-0.3.0-SNAPSHOT.pom  sarif4k-linuxx64-0.3.0-SNAPSHOT-sources.jar

/c/Users/a00665046/.m2/repository/io/github/detekt/sarif4k/sarif4k-mingwx64/0.3.0-SNAPSHOT:
maven-metadata-local.xml  sarif4k-mingwx64-0.3.0-SNAPSHOT.klib  sarif4k-mingwx64-0.3.0-SNAPSHOT.module  sarif4k-mingwx64-0.3.0-SNAPSHOT.pom  sarif4k-mingwx64-0.3.0-SNAPSHOT-sources.jar
```

Reply:
resolved, removed setting artifactId explicitly